src/main/java/org/symphonyoss/symphony/messageml/elements/Radio.java

```diff
@@ -0,0 +1,190 @@
+      case CHECKED_ATTR:
+        setAttribute(CHECKED_ATTR, getStringAttribute(item));
+        break;
+      default:
+        throw new InvalidInputException("Attribute \"" + item.getNodeName()
+            + "\" is not allowed in \"" + getMessageMLTag() + "\"");
+    }
+  }
+
+  private void buildElementFromGroupDiv(MessageMLParser context, org.w3c.dom.Element element) throws InvalidInputException, ProcessingException {
+    NodeList children = element.getChildNodes();
+
+    if (children.getLength() != PRESENTATIONML_DIV_NUMBER_OF_CHILDREN) {
+      throw new InvalidInputException(String.format("Invalid PresentationML for the \"%s\" element", MESSAGEML_TAG));
+    }
+
+    String firstNodeName = "";
+    for (int i = 0; i < PRESENTATIONML_DIV_NUMBER_OF_CHILDREN; i++) {
+      if(firstNodeName.equals(children.item(i).getNodeName())) {
+        throw new InvalidInputException(String.format("Invalid PresentationML for the \"%s\" element", MESSAGEML_TAG));
+      }
+
+      switch (children.item(i).getNodeName()) {
+        case INPUT_TAG:
+          buildAttrFromInputTag(children.item(i));
+          firstNodeName = INPUT_TAG;
+          break;
+        case PRESENTATIONML_LABEL_TAG:
+          buildTextFromLabelTag(context, children.item(i));
+          firstNodeName = PRESENTATIONML_LABEL_TAG;
+          break;
+        default:
+          throw new InvalidInputException(String.format("Invalid PresentationML for the \"%s\" element", MESSAGEML_TAG));
+      }
+    }
+  }
+
+  private void buildTextFromLabelTag(MessageMLParser context, org.w3c.dom.Node labelElement) throws InvalidInputException, ProcessingException {
+    NodeList childNodes = labelElement.getChildNodes();
+    if(childNodes == null || childNodes.getLength() <= 0) {
+      throw new InvalidInputException(String.format("Invalid PresentationML for the \"%s\" element", MESSAGEML_TAG));
+    }
+
+    for (int i = 0; i < childNodes.getLength(); i++) {
+      buildNode(context, childNodes.item(i));
+    }
+  }
+
+  private void buildAttrFromInputTag(org.w3c.dom.Node inputElement) throws InvalidInputException {
+    NamedNodeMap inputAttributes = inputElement.getAttributes();
+    inputAttributes.removeNamedItem(TYPE_ATTR);
+
+    for (int i = 0; i < inputAttributes.getLength(); i++) {
+      buildAttribute(inputAttributes.item(i));
+    }
+  }
+
+  private Map<String, String> buildRadioInputAttributes() {
+    Map<String, String> presentationAttrs = new LinkedHashMap<>();
+    presentationAttrs.put(TYPE_ATTR, PRESENTATIONML_INPUT_TYPE);
+    presentationAttrs.put(NAME_ATTR, getAttribute(NAME_ATTR));
+
+    if (getAttribute(VALUE_ATTR) != null) {
+      presentationAttrs.put(VALUE_ATTR, getAttribute(VALUE_ATTR));
+    } else {
+      presentationAttrs.put(VALUE_ATTR, PRESENTATIONML_DEFAULT_RADIO_VALUE);
+    }
+
+    if (getAttribute(CHECKED_ATTR) != null) {
+      presentationAttrs.put(CHECKED_ATTR, getAttribute(CHECKED_ATTR));
+    }
+
+    return presentationAttrs;
+  }
+}
```

Comment:
If you're going to update this file please consider adding a line break.

Reply:
added a line break